File: hospital-be-main/src/HospitalLibrary/Core/DTO/ViewAppointmentByDoctorDTO.cs
﻿
using HospitalLibrary.Core.Enum;
using HospitalLibrary.Core.Model;
using System;


namespace HospitalLibrary.Core.DTO
{
    public class ViewAppointmentByDoctorDTO
    {
        public Guid PatientId { get; set; }
        public Status Status { get; set; }
        public string PatientName { get; set; }
        public Guid AppointmentId { get; set; }
        public DateRange Date { get; set; }

        public ViewAppointmentByDoctorDTO(Appointment app)
        {
            PatientId = app.PatientId;
            Status = app.Status;
            Date = app.Range;
            AppointmentId = app.Id;
            PatientName = app.Patient.Name + ' ' + app.Patient.Surname;
        }
    }
}

File: hospital-fe-main/src/app/modules/hospital/model/viewAppointmentByDoctorDTO.ts
import {DateRange} from "./date-range.model";

export class ViewAppointmentByDoctorDTO {
  patient: string = "";
  date!: DateRange;


  public constructor(Patient: string = "", Date: DateRange
) {
      this.patient = Patient;
      this.date = Date;

  }
}
